staging/src/k8s.io/kubectl/pkg/cmd/create/create.go

```diff
@@ -65,7 +192,7 @@ type CreateOptions struct {
 	Raw              string
 
 	Recorder genericclioptions.Recorder
-	PrintObj func(obj kruntime.Object) error
+	Printer  printers.ResourcePrinter
```

Comment:
What is the reason of this change in terms of options-flag split?

Reply:
I was following `wait` code, and I saw that it's using the generic `printers.ResourcePrinter` instead of `func(obj kruntime.Object) error`, so I looked into what the type definition of `printers.ResourcePrinter` 
```go
type ResourcePrinter interface {
	// PrintObj receives a runtime object, formats it and prints it to a writer.
	PrintObj(runtime.Object, io.Writer) error
}
```

so I guess `ResourcePrinter` is an almost similar but better options since it will take in an `io.Writer`, and you can write  to different output stream